util/Migrator/DbScripts/2023_01-15_00_KDFOptions.sql

```diff
@@ -0,0 +1,251 @@
+IF COL_LENGTH('dbo.User', 'KdfMemory') IS NULL
```

Comment:
I don't think this migration script is following evolutionary database design principles, but will let some others comment on what should be done to improve.

Reply:
I'll admit, I'm not too practiced with SQL databases or EDD. I tried looking at the existing migration scripts, and basing this one on them.

I'll document my thought process a bit: After reading through the [documentation](https://contributing.bitwarden.com/getting-started/server/database/mssql/edd/) on EDD in Bitwarden: 

One example given for migration in the docs is this:
```sql
-- Add Column
IF COL_LENGTH('[dbo].[Customer]', 'FirstName') IS NULL
BEGIN
    ALTER TABLE
        [dbo].[Customer]
    ADD
        [FirstName] NVARCHAR(MAX) NULL
END
GO
```
In an existing migration script we can find this: https://github.com/bitwarden/server/blob/88b5837baeb2af1787231ebefd5d64482a2f4476/util/Migrator/DbScripts/2023-01-06_00_SecretsManager.sql#L1-L8

So I believe the way the new columns are added in this pull-request is the same as shown in the documentation.
On the other hand I'm not sure if it is compatible since this would break the existing stored procedures.

The migration script does also update the stored procedures, but In a non backward-compatible way.

In an existing migration script we have:
https://github.com/bitwarden/server/blob/88b5837baeb2af1787231ebefd5d64482a2f4476/util/Migrator/DbScripts/2023-01-06_00_SecretsManager.sql#L68-L215
Which seems to deal with the issue by  setting a default value for "UseSecretsManager".

So is setting a default value for KdfMemory and KdfParallelism enough to ensure following EDD, or is there anything else required that I can work on?

Reply:
Improved this migration script a bit, still need to test whether the rollback is possible without issues.